SIAB/spillage/torchutils.py

```
@@ -0,0 +1,168 @@
+    the coef here is flattened, unlike what in the original impl.
+    coef is indexed by [itype, l, zeta, q] -> float
+
+    natom still in type of list[int] and nbes still list[list[int]],
+    indexed by [itype, l] -> int
+    '''
+    assert len(natom) == len(nbes) # == len(coef) # ntype equal, but coef is flattened
+
+    def _gen_q2zeta(coef, natom, nzeta, nbes):
+        lmax = [len(nbes_t) - 1 for nbes_t in nbes]
+        index_, range_ = _coef_flatten(nzeta, nbes)
+        for itype, _, l, _ in _lin2comp(natom, lmax=lmax):
```

Ruff

Ambiguous variable name: `l`

(E741)